Rename SMTChecker to SolCMC across codebase and docs

furmak331 opened the pull request:
I've completed the renaming of `SMTChecker` to `SolCMC` across the codebase and documentation.
It was mentioned in the issue  #15875. 
The build completes successfully after these changes.
Let me know if its properly done .
Please review and let me know if anything needs to be adjusted. Thanks!

blishko commented:
There is something wrong with your changes.
It seems you are changing something in pretty much every file in the codebase, which is certainly not correct.
Please, try to have a look and only modify what really needs to be modified.

furmak331 commented:
sorry @blishko 
You're right . it looks like a large number of files were marked as modified, even though I didn’t intend to change their content. this probably happened because of line ending conversion (LF vs CRLF) on my system.Maybe git automatically changed the line endings to CRLF when I touched those files locally, which made it appear as though they were modified.
I'll fix this by configuring Git to use the correct line endings and resubmit the PR with only the necessary changes.

blishko commented:
Indeed this looks like line ending issue.

furmak331 commented:
just corrected it the line endings but it seems that there seems much more occurences than what i have changed.
Please check the image attached which displays around 1.5K files .Should i change them all to SolCMC?
![Screenshot 2025-04-25 110606](https://github.com/user-attachments/assets/82c2910e-e582-48b2-a755-cc579cd23c2a)

blishko commented:
The diff now seems much more reasonable. I am not sure what are the 1.5k files you mention in the screenshot.
Even with your current proposed changes, not all of them should be changed. For example the name should not be changed in the Changelog.
Similarly, the change in pragma is not this straightforward. We need to discuss how exactly we should proceed here.
Overall, this change needs more methodical approach, it is not a simple renaming.

furmak331 commented:
Yes i should have done it properly rather than renaming randomly .  i did it in a bit hurry and started renaming.
I should rename where it is semantically correct (e.g., class names, file names related to the checker itself).
i will avoid changing occurrences like Changelog history or unrelated comments.
and have to be careful with pragma or public-facing references where backward compatibility or historical correctness is important.
Is that how we should proceed ?
Please guide me a bit regarding this.

blishko commented:
You can start with renaming in the documentation. That should be safe.
As I said, we have to discuss the implication with respect to the existing pragma.
This change was meant for 0.9., which is still some time away.

furmak331 committed:
rename the instances of smtchecker to solCMC across documentation

furmak331 committed:
Merge branch 'develop' into rename-smtchecker-to-solcmc

furmak331 commented:
For now ,I’ve renamed all occurrences of  smtchecker in the documentation files under the docs/ directory.
I’ve avoided touching any code files or pragma changes for now, as discussed. Please let me know if any adjustments are needed.

furmak331 committed:
Fix undefined label reference for 'solcmc' in security-considerations.rst

furmak331 committed:
Merge branch 'rename-smtchecker-to-solcmc' of https://github.com/furmak331/solidity into rename-smtchecker-to-solcmc

furmak331 committed:
Fix undefined label reference for 'solcmc' in security-considerations.rst

furmak331 committed:
Update security-considerations.rst

furmak331 committed:
Update security-considerations.rst

furmak331 committed:
revert back changes

blishko commented:
Hi @furmak331 , after some discussion, the situation is as follows:
To prevent unnecessary inconsistencies, we cannot rename `SMTChecker` before we can drop the `SMTChecker` pragma.
And that is a breaking change, that's why it has been planned for `0.9`.

So this has to be a two-step change (dropping pragma and then renaming) and it cannot be done at the moment.

furmak331 commented:
Thanks @blishko  for explaining! I understand now why the rename can't be done immediately.We can close this PR for now right? I'll look for other issues to work on.
Would you recommend any other open issues ?

blishko commented:
Yes, I suggest to close this one for now.
I don't have any recommendations at the moment, unfortunately.
